d2ir/d2ir.go

```diff
@@ -663,29 +679,73 @@ func (m *Map) EnsureField(kp *d2ast.KeyPath, refctx *RefContext) (*Field, error)
 		}
 		i++
 	}
-	return m.ensureField(i, kp, refctx)
+
+	var fa []*Field
+	err := m.ensureField(i, kp, refctx, create, &fa)
+	return fa, err
 }
 
-func (m *Map) ensureField(i int, kp *d2ast.KeyPath, refctx *RefContext) (*Field, error) {
+func (m *Map) ensureField(i int, kp *d2ast.KeyPath, refctx *RefContext, create bool, fa *[]*Field) error {
+	us, ok := kp.Path[i].Unbox().(*d2ast.UnquotedString)
+	if ok && us.Pattern != nil {
+		fa2, ok := m.doubleGlob(us.Pattern)
```

Comment:
i wonder if in the parser we should restrict double globs to just be on their own. here i was testing what double glob behavior would be with prepending or appending something, but i realize it actually just treated this as two separate globs, which is meaningless. 

```d2
fast: {
  a
  far
}

task: {
  a
}

f**.style.fill: red
```

![Screen Shot 2023-07-30 at 11 24 23 AM](https://github.com/terrastruct/d2/assets/3120367/f2e28f34-0aab-4dc4-8766-8d31ba46d3a7)

Reply:
Not sure what you mean. d2ir can use the parsed pattern from the parser to detect the kind of double glob you've given an example of above, it just doesn't mean anything beyond a single glob right now.